core/backend/src/IModelDb.ts

```diff
@@ -606,6 +608,32 @@ export abstract class IModelDb extends IModel {
     return stmt;
   }
 
+  /**
+   * queries the BisCore.SubCategory table for the entries that are children of the passed categoryIds
+   * @param categoryIds categoryIds to query
+   * @returns array of SubCategoryResultRow
+   */
+  public async querySubCategories(categoryIds: Id64String[]): Promise<SubCategoryResultRow[]> {
```

Comment:
Is this intentionally `@public`? `SubCategoryResultRow` is `@internal`.

Reply:
`SubCategoryResultRow` is redundant with `Categories.SubCategoryInfo`, which was [introduced](https://github.com/iTwin/itwinjs-core/pull/3893) after you first submitted this PR. Except the latter is defined in core-frontend.
If you intend to include this in 3.3.0 we should consolidate the two and move `SubCategoryInfo` (and I guess `CategoryInfo`) to core-common.

Reply:
Since Categories.SubCategoryInfo has the class SubCategoryAppearance as opposed to the Props (from SubCategoryResultRow)...

Is it fine to pass the whole class back to the frontend? Or should Categories.SubCategoryInfo be changed to have the props as the appearance instead of the class?

Reply:
Oh, good catch. Anything we send over RPC needs to be a plain-old JavaScript object.

Still trying to find out if you intend to make all this `@public`. If so maybe `SubCategoryInfoProps` is a better name than `SubCategoryResultRow`.

Reply:
I have no preference either way. I'm guessing it should be public since I originally made this PR with OCP and DR in mind.

Reply:
I also changed it so that we're only using SubCategoryResultRow in viewstatehydrator to support older frontends. And I made Categories.SubCategoryInfo's appearance prop be the SubCategoryAppearance.Props instead of the entire class. 

open to suggestions there... Your comment about SubCategoryInfoProps makes me think you did not want that though. So maybe we should keep Categories.SubCategoryInfo as is, rename subcategoryresultrow and keep the props of that interface "subcategoryresultrow" the same?

Reply:
No, I definitely want users of the Categories API to get back instances of SubCategoryAppearance, not Props. Frankly, I don't think `IModelDb.querySubCategories` needs to be `@public`; it exists only to serve the RPC interface. I would move its logic inside the RPC impl or make the method `@internal`. The frontend provides the Categories API to efficiently query (possibly-cached) subcategory info using the RPC interface under the hood; backend code needs no such API because there's no need for caching and ECSql works just as well.